src/app/modules/main/wallet_section/new_tokens/token_by_symbol_model.nim

```diff
@@ -0,0 +1,111 @@
+import NimQml, Tables, strutils
+
+import ./io_interface
+import ../../../../core/eventemitter
+
+type
+  ModelRole {.pure.} = enum
+    Key = UserRole + 1
+    Name
+    Symbol
+    # uniswap/status/custom seq[string]
+    Sources
+    AddressPerChain
+    Decimals
+    Image
+    # Native, Erc20, Erc721
+    Type
+    # only be valid if source is custom
+    CommunityId
+    # everything below should be lazy loaded
+    Description
+    # properties below this are optional and may not exist in case of community minted assets
+    # built from chainId and address using networks service
+    WebsiteUrl
+    MarketValues
+
+QtObject:
+  type TokensBySymbolModel* = ref object of QAbstractListModel
```

Comment:
why is it named token by symbol, i would expect a map in this case?

Reply:
That means the `key`, or the rule used for grouping and building the list is `by symbol`. However it's not 100% like this since, when they are **community tokens**, there are some corner cases where `symbol` **can not be unique** that's why here it is introduced the following exception: 
- If `communitiId != null` --> `key` will be the token `address`.

Reply:
**Example**:
`dai symbol` can be part of different sources: `uniswap` and `status` sources:

1. In the _flat list_, there will be 2 different entries for `dai symbol`, and `key` for each item will be the pair **<address, chainId>.**
2. In the _token by symbol list_, there will be only 1 entry for `day symbol`, and `key` for each item will be in general the **symbol** (taking into account there is an exception with community token items as explained before).